### PR TITLE
Added unittests for experiments

### PR DESCRIPTION
**Pull request by ankushjain2001**
Please review these but do not merge this PR yet as it will fail due to library incompatibility issues. I am sharing this to give you an idea of the kind of test cases I'm making. 

For now, I first need to work on sorting out the libraries/dependencies and then I will start improving the test cases.

### CODE REVIEW

File: fp/tests/test_experiments.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
+import os
+import unittest
+import pickle
+from pandas import read_csv
+from datetime import datetime
+#from freezegun import freeze_time
```
Comment (sscdotopen) [inline fp/tests/test_experiments.py:6]: please remove the commented code

File: fp/tests/test_experiments.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+from fp.scalers import NoScaler
+from fp.learners import NonTunedLogisticRegression, NonTunedDecisionTree
+from fp.pre_processors import NoPreProcessing
+from fp.post_processors import NoPostProcessing
+from fp.experiments import BinaryClassificationExperiment
+
+
+#@freeze_time('2020-01-01 00:00:00.000000')
+class testSuiteExperiments(unittest.TestCase):
+
```
Comment (sscdotopen) [inline fp/tests/test_experiments.py:22]: we dont need three blank lines here

File: fp/tests/test_experiments.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+from fp.traindata_samplers import CompleteData
+from fp.missingvalue_handlers import CompleteCaseAnalysis
+from fp.scalers import NoScaler
+from fp.learners import NonTunedLogisticRegression, NonTunedDecisionTree
+from fp.pre_processors import NoPreProcessing
+from fp.post_processors import NoPostProcessing
+from fp.experiments import BinaryClassificationExperiment
+
+
+#@freeze_time('2020-01-01 00:00:00.000000')
```
Comment (sscdotopen) [inline fp/tests/test_experiments.py:20]: remove pls

File: fp/tests/test_experiments.py [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+                                                         label_name, positive_label, numeric_attribute_names,
+                                                         categorical_attribute_names, attributes_to_drop_names,
+                                                         train_data_sampler, missing_value_handler, numeric_attribute_scaler,
+                                                         learners, pre_processors, post_processors,
+                                                         protected_attribute_names, privileged_classes, privileged_groups,
+                                                         unprivileged_groups, dataset_metadata, dataset_name)
+        self.data = read_csv('fp/tests/resource/input/data.csv')
+        self.annotated_train_data = pickle.load(open('fp/tests/resource/input/data_annotated.obj', 'rb'))
+    
+    # Test 01 - Validate __init__
```
Comment (sscdotopen) [inline fp/tests/test_experiments.py:65]: remove pls

File: fp/tests/test_experiments.py [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+        self.assertEqual(self.experiment.privileged_groups, [{'sex': 1.0}])
+        self.assertEqual(self.experiment.unprivileged_groups, [{'sex': 0.0}])
+        self.assertEqual(self.experiment.dataset_metadata, {'label_maps': [{1.0: 1, 0.0: 0}], 
+                                                            'protected_attribute_maps': [{1.0: 'male', 0.0: 'female'}]
+                                                            })
+        self.assertEqual(self.experiment.dataset_name, 'test_dataset')
+        self.assertEqual(self.experiment.log_path, 'logs/')
+        self.assertEqual(self.experiment.exec_timestamp, datetime.now().strftime('%Y-%m-%d_%H-%M-%S-%f')[:-3])
+
+    # Test 02 - Validate unique_file_name
```
Comment (sscdotopen) [inline fp/tests/test_experiments.py:100]: remove comment

File: fp/tests/test_experiments.py [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+                                                         self.experiment.pre_processors[0], 
+                                                         self.experiment.post_processors[0]),
+            'test_dataset__LogisticRegression-notuning__complete_case__complete_data__no_scaler__no_pre_processing__no_post_processing')
+        self.assertEqual(self.experiment.unique_file_name(self.experiment.learners[1], 
+                                                         self.experiment.pre_processors[0], 
+                                                         self.experiment.post_processors[0]),
+            'test_dataset__DecisionTree-notuning__complete_case__complete_data__no_scaler__no_pre_processing__no_post_processing')
+
+
+    # Test 03 - Validate generate_file_path
```
Comment (sscdotopen) [inline fp/tests/test_experiments.py:112]: remove pls

**Comment by sscdotopen** [comment]
lgtm, please remove the commented out code and make sure that the build does not fail

**Comment by ankushjain2001** [comment]
> lgtm, please remove the commented out code and make sure that the build does not fail

I have fixed the script as per your comments. I'll need some time to figure out the dependency mismatch issue with Travis. The failure is not related to this code as it works perfectly in local environment. It's related to Travis virtual test env and dependencies.

**Pull request closed without merging**